thriftbp/headers.go

```diff
@@ -44,3 +45,27 @@ func AddClientHeader(ctx context.Context, key, value string) context.Context {
 	headers = append(headers, key)
 	return thrift.SetWriteHeaderList(ctx, headers)
 }
+
+// Header gets the value of a thrift header by key
+//
+// If the value is not present we fall back to a case-insensitive check
+func Header(ctx context.Context, key string) (v string, ok bool) {
+
+	// Option 1.
+	v, ok = thrift.GetHeader(ctx, key)
+	if !ok {
+		v, ok = thrift.GetHeader(ctx, strings.ToLower(key))
```

Comment:
if you are certain that what envoy does is just lowercase the key, then this is enough and we won't need option 2.